features/networking/service.feature

```diff
@@ -153,3 +153,334 @@ Feature: Service related networking scenarios
+    """
+    
+    # Create a svc with externalIP
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/externalip_service1.json" replacing paths:
+      | ["spec"]["externalIPs"][0]  | <%= cb.host2ip %> |
+    Then the step should succeed
+    Then I wait up to 10 seconds for the steps to pass:
+    """
+    When I run the :get client command with:
+      | resource      | service               |
+      | resource_name | service-unsecure      |
+      | template      | {{.spec.externalIPs}} |
+      | o             | yaml                  |
+    And the output should contain "<%= cb.host2ip %>"
+    """
+ 
+    # Create a pod
+    When I run the :create client command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/routing/caddy-docker.json |
+    Then the step should succeed
+    And the pod named "caddy-docker" becomes ready
+ 
+    # Curl externalIP:portnumber on new pod 
+    When I execute on the pod:
+      | /usr/bin/curl | -k | <%= cb.host2ip %>:27017 |
+    Then the output should contain:
+      | Hello-OpenShift-1 http-8080                  |
+
+   
+  # @author weliang@redhat.com
+  # @case_id OCP-24692
+  @admin
+  @destructive
+  Scenario: A rejectedCIDRs inside an allowedCIDRs	 
+    # Create additional network through CNO
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"allowedCIDRs":["22.2.2.0/24"],"rejectedCIDRs":["22.2.2.0/25"]}}}} |
+    Then I wait up to 10 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource      | networks.config.openshift.io                 |
+      | resource_name | cluster                                      |
+      | template      | {{{{.spec.externalIP.policy.allowedCIDRs}}}} |
+      | o             | yaml                                         |
+    And the output should contain "22.2.2.0/24"
+    When I run the :get admin command with:
+      | resource      | networks.config.openshift.io                  |
+      | resource_name | cluster                                       |
+      | template      | {{{{.spec.externalIP.policy.rejectedCIDRs}}}} |
+      | o             | yaml                                          |
+    And the output should contain "22.2.2.0/25"
+    """
+    #Clean-up required to erase above net-attach-def after testing done
+    Given I register clean-up steps:
+    """
+    Given as admin I successfully merge patch resource "networks.config.openshift.io/cluster" with:
+      | {"spec":{"externalIP":{"policy":{"allowedCIDRs":null }}}}  |
+    """
+
+    # Need wait for about 500 seconds to apply new additional network configuration
+    # In the future this script can be updated if we find a way to check whether cluster is reay with new CNO configuration and not using 500 seconds
+    Given 500 seconds have passed
```

Comment:
@weliang1 I understand we already had discussion on this in past and i agree it takes time. But just curious, do you see any userspaceproxier logs in  network operator pod which confirms this change has been effective? If yes, we can implement a polling logic here. If not we can stay happy with exisisiting :)

Reply:
@anuragthehatter Agree, continue working on it.

Reply:
@anuragthehatter Using polling method instead of sleeping, thanks! New test logs: 
http://pastebin.test.redhat.com/828370
http://pastebin.test.redhat.com/828372
http://pastebin.test.redhat.com/828378
https://privatebin-it-iso.int.open.paas.redhat.com/?3e61cb0b743717f6#C51i1j0ryAnz4hn0ZN/g2DtEUaVO8ClxFlgNR7ByedI=
http://pastebin.test.redhat.com/828389
http://pastebin.test.redhat.com/828392

Reply:
@anuragthehatter @pruan-rht @zhaozhanqi @huiran0826 @rbbratta please let me know whether the script still need improvements, thanks!

Reply:
Thanks @weliang1. Would have been great efforts testing this. Looks good for now. We need to poll CNO or SDN logs to track when a change has been in effect instead instead of creating service numerous times to wait for a pass one. If logs ar enot sufficently telling, then we can open a bug for improved logging.

Reply:
@anuragthehatter Submitted a bug to track this: Bug 1793099 - Modifying ExternalIP policy in CNO take about 220 seconds to take effect